packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -141,44 +121,38 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
   private ddbTableMap: Record<string, ITable> = {};
   private resolverMap: Record<string, TransformerResolverProvider> = {};
   private typesWithModelDirective: Set<string> = new Set();
+  private resourceGeneratorMap: Map<string, ModelResourceGenerator> = new Map<string, ModelResourceGenerator>();
+  private modelToDatasourceMap: Map<string, DatasourceType> = new Map<string, DatasourceType>()
   /**
    * A Map to hold the directive configuration
    */
   private modelDirectiveConfig: Map<string, ModelDirectiveConfiguration> = new Map();
   constructor(options: ModelTransformerOptions = {}) {
     super('amplify-model-transformer', directiveDefinition);
     this.options = this.getOptions(options);
+    const rdsGenerator = new RdsModelResourceGenerator();
+    this.resourceGeneratorMap.set(DDB_DB_TYPE, new DynamoModelResourceGenerator());
+    this.resourceGeneratorMap.set(MYSQL_DB_TYPE, rdsGenerator);
   }
 
   before = (ctx: TransformerBeforeStepContextProvider): void => {
-    // add model related-parameters to the root stack
-    ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBModelTableReadIOPS, {
-      description: 'The number of read IOPS the table should support.',
-      type: 'Number',
-      default: 5,
-    });
-    ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBModelTableWriteIOPS, {
-      description: 'The number of write IOPS the table should support.',
-      type: 'Number',
-      default: 5,
-    });
-    ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBBillingMode, {
-      description: 'Configure @model types to create DynamoDB tables with PAY_PER_REQUEST or PROVISIONED billing modes.',
-      default: 'PAY_PER_REQUEST',
-      allowedValues: ['PAY_PER_REQUEST', 'PROVISIONED'],
-    });
-    ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBEnablePointInTimeRecovery, {
-      description: 'Whether to enable Point in Time Recovery on the table.',
-      type: 'String',
-      default: 'false',
-      allowedValues: ['true', 'false'],
-    });
-    ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBEnableServerSideEncryption, {
-      description: 'Enable server side encryption powered by KMS.',
-      type: 'String',
-      default: 'true',
-      allowedValues: ['true', 'false'],
-    });
+    const datasourceMapValues: Array<DatasourceType> = Array.from(ctx.modelToDatasourceMap.values());
+    if (datasourceMapValues.some((value) => value.dbType === DDB_DB_TYPE && value.provisionDB)) {
+      this.resourceGeneratorMap.get(DDB_DB_TYPE)?.enableGenerator();
+      this.resourceGeneratorMap.get(DDB_DB_TYPE)?.enableProvisioned();
+    }
+    if (datasourceMapValues.some((value) => value.dbType === MYSQL_DB_TYPE && !value.provisionDB)) {
+      this.resourceGeneratorMap.get(MYSQL_DB_TYPE)?.enableGenerator();
+      this.resourceGeneratorMap.get(MYSQL_DB_TYPE)?.enableUnprovisioned();
```

Comment:
Why do we need these steps? I assume we have to enable these all times.

Reply:
I did this just to prevent unnecessary resource creation. If for example, there are no RDS models, this allows us to not bother creating the lambda or anything. And the resources we create for provisioned will be different than unprovisioned